internal/controllers/topology/cluster/cluster_controller.go

```diff
@@ -336,6 +338,36 @@ func (r *Reconciler) machineDeploymentToCluster(o client.Object) []ctrl.Request
 	}}
 }
 
+func (r *Reconciler) reconcileDelete(ctx context.Context, cluster *clusterv1.Cluster) (ctrl.Result, error) {
+	/*
+		If the RuntimeSDK feature gate is enabled and the cluster has the BeforeClusterDelete hook as pending
+		then call the BeforeClusterDelete hook.
+
+		If the result of the hook is a non-blocking success, mark the hook as done we are done.
+		If the result of teh hook is a blocking success, simply return a result with retry after
+		If we get an error do the general error handling.
+	*/
+	if feature.Gates.Enabled(feature.RuntimeSDK) {
+		if hooks.IsPending(runtimehooksv1.BeforeClusterDelete, cluster) {
+			hookRequest := &runtimehooksv1.BeforeClusterDeleteRequest{
+				Cluster: *cluster,
+			}
+			hookResponse := &runtimehooksv1.BeforeClusterDeleteResponse{}
+			if err := r.RuntimeClient.CallAllExtensions(ctx, runtimehooksv1.BeforeClusterDelete, cluster, hookRequest, hookResponse); err != nil {
+				return ctrl.Result{}, err
+			}
+			if hookResponse.RetryAfterSeconds != 0 {
+				return ctrl.Result{RequeueAfter: time.Duration(hookResponse.RetryAfterSeconds) * time.Second}, nil
+			}
+			// The BeforeClusterDelete hook returned a non-blocking response. We can remove this hook from the list of pending-hooks.
+			if err := hooks.MarkAsDone(ctx, r.Client, cluster, runtimehooksv1.BeforeClusterDelete); err != nil {
+				return ctrl.Result{}, errors.Wrapf(err, "failed to remove the %s hook from pending hooks tracker", runtimecatalog.HookName(runtimehooksv1.BeforeClusterDelete))
```

Comment:
```suggestion
				return ctrl.Result{}, err
```
Can we make sure to build this error inside of the func? (please also check for other call sites)